### ISSUE TITLE
Compiling transformation error on save 

### ISSUE THREAD

**Issue by MorithK** [ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator#478]
Good evening, tried to upgrade my project tonight from a version that was a year or so old but am getting errors. Thanks in advance. Victoria

Win 7, VS 2012, EF: 6.2.0 from NuGet, .Net 4.5 
Reverse POCO Generator: 2.37.2 (updated from Extensions and Updates in VS)
I have installed Entity Framework 6 Tools from link on front page. (EFTools6.1.3ForVS2012.msi)

Getting this compilation error on saving the Database.tt
![image](https://user-images.githubusercontent.com/20152712/49198674-e8500f80-f359-11e8-9fac-7f4e2b284856.png)

EF.Reverse.POCO.Core.ttinclude Lines 4275 - 4289:

`            sb.AppendLine(
                $"            var {WriteStoredProcSqlParameterName(p)} = new System.Data.SqlClient.SqlParameter"
                + $" {{ ParameterName = \"{p.Name}\", "
                + (isGeography ? "UdtTypeName = \"geography\"" : $"SqlDbType = System.Data.SqlDbType.{p.SqlDbType}")
                + ", Direction = System.Data.ParameterDirection."
                + (p.Mode == StoredProcedureParameterMode.In ? "Input" : "Output")
                + (p.Mode == StoredProcedureParameterMode.In
                    ? ", Value = " + (isGeography
                        ? $"Microsoft.SqlServer.Types.SqlGeography.Parse({p.NameHumanCase}.AsText())"
                        : p.NameHumanCase + getValueOrDefault)
                    : string.Empty)
                + (p.MaxLength != 0 ? ", Size = " + p.MaxLength : string.Empty)
                + ((p.Precision > 0 || p.Scale > 0) ? ", Precision = " + p.Precision + ", Scale = " + p.Scale : string.Empty)
                + (p.PropertyType.ToLower().Contains("datatable") ? ", TypeName = \"" + p.UserDefinedTypeName + "\"" : string.Empty)
                + " };");
`

**Comment by MorithK** [comment]
Just to be sure this wasn't any lurking code I made a brand new project in a new solution and I'm getting the same error.

**Comment by sjh37** [comment]
Your project must be using C# 5. Can you update your project to C# 6 or even 7?

I try to keep the project to C# 5 so it has the widest possibility of use, and I thought I got rid of all the C# 6 `Interpolated Strings` but I can see there is still one there. Sorry.

You will have to convert this to a String.Format(...) instead of the $"..." interpolated string format and all should be ok.

**Comment by sjh37** [comment]
Try changing it to 

```c#
sb.AppendLine(
                string.Format("            var {0} = new System.Data.SqlClient.SqlParameter", WriteStoredProcSqlParameterName(p))
                + string.Format(" {{ ParameterName = \"{0}\", ", p.Name)
                + (isGeography ? "UdtTypeName = \"geography\"" : string.Format("SqlDbType = System.Data.SqlDbType.{0}", p.SqlDbType))
                + ", Direction = System.Data.ParameterDirection."
                + (p.Mode == StoredProcedureParameterMode.In ? "Input" : "Output")
                + (p.Mode == StoredProcedureParameterMode.In
                    ? ", Value = " + (isGeography
                        ? string.Format("Microsoft.SqlServer.Types.SqlGeography.Parse({0}.AsText())", p.NameHumanCase)
                          : p.NameHumanCase + getValueOrDefault)
                    : string.Empty)
                + (p.MaxLength != 0 ? ", Size = " + p.MaxLength : string.Empty)
                + ((p.Precision > 0 || p.Scale > 0) ? ", Precision = " + p.Precision + ", Scale = " + p.Scale : string.Empty)
                + (p.PropertyType.ToLower().Contains("datatable") ? ", TypeName = \"" + p.UserDefinedTypeName + "\"" : string.Empty)
                + " };");

                if (p.Mode == StoredProcedureParameterMode.In)
                {
                    sb.AppendFormat(
                        isNullable
                            ? "            if (!{0}.HasValue){1}                {0}Param.Value = System.DBNull.Value;{1}{1}"
                            : "            if ({0}Param.Value == null){1}                {0}Param.Value = System.DBNull.Value;{1}{1}",
                        p.NameHumanCase, Environment.NewLine);
                }
```

**Comment by MorithK** [comment]
I do not appear to be able to go above C#5 in VS2012 unless I am missing something on where to upgrade that. 

Thank you, the string change worked on the previous errors but two more showed up:
![image](https://user-images.githubusercontent.com/20152712/49307848-0cb40500-f49c-11e8-81c3-d4b2f3eaf7b2.png)

I am attempting the rabbit hole of upgrading the whole project to VS2017 community edition (reverse poco is working great there but I have a whole different set of issues) but figured you might want to be aware of the above if you wish to continue supporting 2012. I'm also picking up an 6yr old project and trying to relearn entity framework and linq to sql after not touching it in quite a while so it's been a learning experience. My day job is almost purely SQL management studio.

**Comment by sjh37** [comment]
Just replace `nameof(xxx)` with `"xxx"`. See latest changeset for the update.

with regard to `isGeographys`, I can only see `isGeography` in the code did you accidentally add an s onto the end of `isGeography` ?

**Comment by MorithK** [comment]
Oops.. yes, that must have been a mistype on my end. 

The other change is perfect! Thank you. Everything is now generating correctly.

**Comment by sjh37** [comment]
This has now been released in v2.37.3
